src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/BaseSyncStrategy.java

```diff
@@ -103,6 +108,10 @@
      */
     AtomicBoolean executing = new AtomicBoolean(false);
 
+    Semaphore exec;
```

Comment:
nit: Can we rename this to something that makes it more readable?

Reply:
renamed to `criticalExecBlock`